web3swift/src/Account/EthereumAccount.swift

```diff
@@ -9,173 +9,107 @@ import Foundation
 public protocol EthereumAccountProtocol {
     var address: EthereumAddress { get }
 
-    func sign(data: Data) throws -> Data
-    func sign(hash: String) throws -> Data
-    func sign(hex: String) throws -> Data
-    func sign(message: Data) throws -> Data
-    func sign(message: String) throws -> Data
-    func sign(transaction: EthereumTransaction) throws -> SignedTransaction
+    func sign(data: Data) async throws -> Data
+    func sign(hash: String) async throws -> Data
+    func sign(hex: String) async throws -> Data
+    func sign(message: Data) async throws -> Data
+    func sign(message: String) async throws -> Data
+    func sign(transaction: EthereumTransaction) async throws -> SignedTransaction
 }
 
 public enum EthereumAccountError: Error {
+    case addressesDoesNotMatch
     case createAccountError
     case importAccountError
     case loadAccountError
     case signError
 }
 
 public class EthereumAccount: EthereumAccountProtocol {
-    private let privateKeyData: Data
-    private let publicKeyData: Data
+    private let keyStorage: EthereumKeyStorageProtocol
     private let logger: Logger
 
-    public lazy var publicKey: String = self.publicKeyData.web3.hexString
+    public let publicKey: String
+    public let address: EthereumAddress
 
-    public lazy var address: EthereumAddress = KeyUtil.generateAddress(from: self.publicKeyData)
-
-    required public init(keyStorage: EthereumSingleKeyStorageProtocol, keystorePassword password: String, logger: Logger? = nil) throws {
+    required public init(address: EthereumAddress, keyStorage: EthereumKeyStorageProtocol, logger: Logger? = nil) {
+        self.keyStorage = keyStorage
         self.logger = logger ?? Logger(label: "web3.swift.eth-account")
-        do {
-            let decodedKey = try keyStorage.loadAndDecryptPrivateKey(keystorePassword: password)
-            self.privateKeyData = decodedKey
-            self.publicKeyData = try KeyUtil.generatePublicKey(from: decodedKey)
-        } catch {
-            self.logger.warning("Error loading key data: \(error)")
-            throw EthereumAccountError.loadAccountError
-        }
+        self.address = address
+        self.publicKey = address.value.web3.withHexPrefix
     }
 
-    required public init(keyStorage: EthereumSingleKeyStorageProtocol, logger: Logger? = nil) throws {
```

Comment:
We should allow for initialization with a KeyStorage as the address may be unknown until private key is loaded